File: src/main/java/miragefairy2019/mod/api/fairycrystal/DropFixed.java
package miragefairy2019.mod.api.fairycrystal;

import net.minecraft.item.ItemStack;

public final class DropFixed implements IDrop {

    public final ItemStack itemStack;
    public final DropCategory dropCategory;
    public final double weight;

    public DropFixed(ItemStack itemStack, DropCategory dropCategory, double weight) {
        this.itemStack = itemStack;
        this.dropCategory = dropCategory;
        this.weight = weight;
    }

    @Override
    public ItemStack getItemStack() {
        return itemStack;
    }

    @Override
    public DropCategory getDropCategory() {
        return dropCategory;
    }

    @Override
    public double getWeight() {
        return weight;
    }

}

File: src/main/java/miragefairy2019/mod3/artifacts/fairycrystal/DropFixed.kt
package miragefairy2019.mod3.artifacts.fairycrystal

import miragefairy2019.mod3.fairy.RankedFairyTypeBundle

class DropFixed(
    val bundle: RankedFairyTypeBundle,
    override val dropCategory: DropCategory,
    override val weight: Double
) : IDrop {
    override fun getItemStack(rank: Int) = bundle[rank].createItemStack()
}
